hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/JavaDefaultSerializers.java

```diff
@@ -76,14 +77,17 @@ public Object read(final ObjectDataInput in) throws IOException {
         }
 
         private Object read(InputStream in, ClassLoader classLoader) throws IOException {
+            ObjectInputStream objectInputStream = null;
             try {
-                ObjectInputStream objectInputStream = newObjectInputStream(classLoader, classFilter, in);
+                objectInputStream = newObjectInputStream(classLoader, classFilter, in);
                 if (shared) {
                     return objectInputStream.readObject();
                 }
                 return objectInputStream.readUnshared();
             } catch (ClassNotFoundException e) {
                 throw new HazelcastSerializationException(e);
+            } finally {
+                closeResource(objectInputStream);
```

Comment:
@Holmistr would not it eventually close the passed `InputStream in`, IIUC, it can be problematic, since in previous implementation `close` was not called, may be it is intentional.